src/classes/CCON_Faculty_TDTM.cls

```diff
@@ -96,29 +95,43 @@ public with sharing class CCON_Faculty_TDTM extends TDTM_Runnable {
                     for (Integer i = 0; i < newlist.size(); i++) {
                         Course_Enrollment__c enroll = (Course_Enrollment__c)newlist[i];
 
-                        if(enroll.get('RecordTypeId') == facultyRecType) {
-                            enroll.Account__c = offerings.get(enroll.Course_Offering__c).Course__r.Account__c;
-
-                            if (contactIdAccountIdAfflList.containsKey(enroll.Contact__c) && (
-                                 contactIdAccountIdAfflList.get(enroll.Contact__c).containsKey(enroll.Account__c))) {
-                                enroll.Affiliation__c = contactIdAccountIdAfflList.get(enroll.Contact__c).get(enroll.Account__c)[0].Id;
-                            } else {
-                                Affiliation__c affl = new Affiliation__c(
-                                    Contact__c = enroll.Contact__c,
-                                    Account__c = enroll.Account__c,
-                                    Role__c = 'Faculty'
-                                );
-                                affiliationsToInsert.add(affl);
-                                connsNeedingAffls.add(enroll);
+                        // Skip over the Course Connection if it is not for a Faculty.
+                        if(enroll.get('RecordTypeId') != facultyRecType) {
+                            continue;
+                        }
+
+                        enroll.Account__c = offerings.get(enroll.Course_Offering__c).Course__r.Account__c;
+
+                        if (contactIdAccountIdAfflList.containsKey(enroll.Contact__c) && (
+                                contactIdAccountIdAfflList.get(enroll.Contact__c).containsKey(enroll.Account__c))) {
+                            enroll.Affiliation__c = contactIdAccountIdAfflList.get(enroll.Contact__c).get(enroll.Account__c)[0].Id;
```

Comment:
I'm normally not a huge fan of chaining so deep. It's rarely ever an issue but easy to lost track of (in my mind) when scanning through code if one isn't super familiar with it.  I feel like we could assign: contactIdAccountIdAfflList.get(enroll.Contact__c) to a variable here to reduce that affect, especially if we are going to use it more than once. Its not a huge deal however.

Reply:
Great suggestion - I'll commit up a quick change momentarily.